var/spack/repos/builtin/packages/xsdktrilinos/package.py

```diff
@@ -0,0 +1,121 @@
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+import os
+import sys
+
+# Trilinos is complicated to build, as an inspiration a couple of links to
+# other repositories which build it:
+# https://github.com/hpcugent/easybuild-easyblocks/blob/master/easybuild/easyblocks/t/trilinos.py#L111
+# https://github.com/koecher/candi/blob/master/deal.II-toolchain/packages/trilinos.package
+# https://gitlab.com/configurations/cluster-config/blob/master/trilinos.sh
+# https://github.com/Homebrew/homebrew-science/blob/master/trilinos.rb and some
+# relevant documentation/examples:
+# https://github.com/trilinos/Trilinos/issues/175
+
+
+class Xsdktrilinos(Package):
+    """The Trilinos Project is an effort to develop algorithms and enabling
+    technologies within an object-oriented software framework for the solution
+    of large-scale, complex multi-physics engineering and scientific problems.
+    A unique design feature of Trilinos is its focus on packages.
+    """
+    homepage = "https://trilinos.org/"
+    base_url = "https://github.com/trilinos/xSDKTrilinos/archive"
+
+    version('develop', git='https://github.com/trilinos/xSDKTrilinos.git', tag='master')
+    version('12.8.1', '9cc338ded17d1e10ea6c0dc18b22dcd4')
+    version('12.6.4', '44c4c54ccbac73bb8939f68797b9454a')
+
+    def url_for_version(self, version):
+        return '%s/trilinos-release-%s.tar.gz' % \
+            (Xsdktrilinos.base_url, version.dashed)
+
+    variant('hypre',        default=True,
+            description='Compile with Hypre preconditioner')
+    variant('petsc',        default=True,
+            description='Compile with PETSc solvers')
+    variant('shared',       default=True,
+            description='Enables the build of shared libraries')
+    variant('debug',        default=False,
+            description='Builds a debug version of the libraries')
+
+    depends_on('cmake', type='build')
+
+    # MPI related dependencies
+    depends_on('mpi')
+    depends_on('hypre', when='+hypre')
+    depends_on('petsc', when='+petsc')
```

Comment:
and since you use `mpi`, you probably want to write `petsc+mpi~complex`. Unless serial version of `petsc` is going to be used in parallel build of `xsdkTrilinos`.